src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -466,21 +466,16 @@ uint32_t PolicyHandler::GetAppIdForSending() const {
   return ChooseRandomAppForPolicyUpdate(apps_with_none_level);
 }
 
-void PolicyHandler::AddNewApplicationIdToPTUQueue(const uint32_t app_id) {
+void PolicyHandler::PushAppIdToQueue(const uint32_t app_id) {
   LOG4CXX_AUTO_TRACE(logger_);
-
-  if (std::find(queue_applications_for_ptu_.begin(),
-                queue_applications_for_ptu_.end(),
-                app_id) == queue_applications_for_ptu_.end()) {
-    queue_applications_for_ptu_.push_back(app_id);
-    policy_manager_->OnChangeApplicationCount(
-        queue_applications_for_ptu_.size());
-  }
+  sync_primitives::AutoLock lock(app_id_queue_lock_);
+  queue_applications_for_ptu_.insert(app_id);
```

Comment:
@ydementieiev check for return result of insert operation. If the same app_id is inserted, count will not be changed

Reply:
@AKalinich-Luxoft done [f0f9ced](https://github.com/smartdevicelink/sdl_core/pull/3208/commits/f0f9ced5f7902b7d25d333c8d1559a7e3f374f89)